test/com/mjcode/model/GameTest.java

```diff
@@ -0,0 +1,277 @@
+
+    }
+    @Test
+    void addPlayer_GameHasOnePlayer() {
+        // Add player 1
+        game.addPlayer(player1);
+
+        // Validate player1 has been added
+        assertEquals(white, player1.getColor());
+        assertEquals(1, player1.getDirection());
+
+        // Add player 2
+        game.addPlayer(player2);
+
+        // Validate player2 has been added
+        assertEquals(black, player2.getColor());
+        assertEquals(-1, player2.getDirection());
+    }
+    @Test
+    void endGame() {
+        // Add the players
+        game.addPlayer(player1);
+        game.addPlayer(player2);
+
+        Piece whitePiece = player1.addPiece(new Tile(3, 2));
+        Piece blackPiece = player2.addPiece(new Tile(6, 1));
+
+        // validate both players has 1 piece in there list
+        assertEquals(1, player1.getNumPieces());
+        assertEquals(1, player2.getNumPieces());
+
+        // Confirm game has not ended since neither player has an empty list
+        assertEquals(1, player1.getNumPieces());
+        assertEquals(1, player2.getNumPieces());
+        assertFalse(game.endGame());
+
+        // Remove piece from player1
+        player1.removePiece(whitePiece);
+
+        // Validate piece has been removed from player1
+        assertEquals(0, player1.getNumPieces());
+        assertTrue(game.endGame());
+    }
+
+    @Test
+    void getTile() {
+        int row = 3;
+        int col = 4;
+
+        Tile expectedTile = new Tile(row, col);
+        board[row][col] = expectedTile;
+
+        Tile actualTile = game.getTile(row, col);
+
+        assertNotNull(expectedTile);
+        assertNotNull(actualTile);
+        assertEquals(expectedTile, actualTile);
+    }
+
+    @Test
+    void executeMove() {
+
+        // Create the tile
+        Tile fromTile = new Tile(2,3);
+        Tile toTile = new Tile(3, 4);
+
+        // Create pieces
+        Piece whitePiece = player1.addPiece(game.getTile(2, 3));
+        Piece blackPiece = player2.addPiece(game.getTile(7, 2));
+
+        // Set the occupant and create a move from tile (2, 3) to (3, 4)
+        fromTile.setOccupant(whitePiece);
+        Move move = new Move(fromTile, toTile);
+
+        // Add the players
+        game.addPlayer(player1);
+        game.addPlayer(player2);
+        // Set the current player
+        game.setCurrentPlayer(player1);
+
+        // Execute the move
+        game.executeMove(move);
+        assertEquals(whitePiece, toTile.getOccupant());
+
+        // Create a new tile from and to tile for the black piece to move
+        Tile secondFromTile = new Tile(7, 2);
+        Tile secondToTile = new Tile(6, 3);
+
+        // Set the occupant and create a move
+        secondFromTile.setOccupant(blackPiece);
+        Move secondMove = new Move(secondFromTile, secondToTile);
+
+        // Execute the second move
+        game.executeMove(secondMove);
+        assertEquals(blackPiece, secondToTile.getOccupant());
+
+    }
+
```

Comment:
Once executeMove is updated with capture, add a test for a piece being captured.